doc/unicode.md

```diff
@@ -1,12 +1,29 @@
 # Lexers and Unicode text
 
-Until ANTLR 4.7, generated lexers only supported part of the Unicode standard
-(code points up to `U+FFFF`).
+Prior to ANTLR 4.7, generated lexers only supported part of the Unicode standard (code points up to `U+FFFF`). As of ANTLR 4.7, the lexers in all language runtimes support the full range of Unicode code points up to `U+10FFFF`, as
+long as the input `CharStream` is opened using `CharStreams.fromPath()`, `CharStreams.fromFileName()`, etc...
+or the equivalent method for your runtime's language. 
 
-With ANTLR 4.7 and later, lexers as well as all languages' runtimes
-support the full range of Unicode code points up to `U+10FFFF`, as
-long as the input `CharStream` is opened using `CharStreams.fromPath()`
-or the equivalent method for your runtime's language.
+The deprecated `ANTLRInputStream` and `ANTLRFileStream` APIs only support Unicode code points up to `U+FFFF`.
+
+A big shout out to Ben Hamilton (github bhamiltoncx) for his superhuman
+efforts across all targets to get true Unicode 3.1 support for U+10FFFF.
```

Comment:
Actually, it was Unicode 2.0 way back in 1996! http://unicode.org/faq/utf_bom.html#gen0